src/wwwroot/controllers/PlanCtrl.js

```diff
@@ -174,6 +184,42 @@
       }      
     }
 
+    function isUpgradeablePlan(limit){
+      //Check constraints to validate that plan is upgradeable
+
+      //Set to true all errors to prevent access if service is not working
+      var errorsPlanUpgrade = {'isDKIMError': true, "isDeliveredError": true, "isDeliveredAndDKIMError": true};
+
+      if(limit==undefined){
+        var isConfigRequired = auth.getLimitsByAccount().then(isUpgradeablePlan).catch($log.error);
+      }
+      else{
+        //Exist activity in the current plan?
+        errorsPlanUpgrade.isDeliveredError = vm.totalDeliveriesCount > 0? false : true;
+        //It is the domain correctly configurated?
+        errorsPlanUpgrade.isDKIMError = limit.requiresDomainConfiguration; 
+        //This account is not registering deliveries and is not correctly configurated?
+        errorsPlanUpgrade.isDeliveredAndDKIMError = errorsPlanUpgrade.isDeliveredError && errorsPlanUpgrade.isDKIMError;
+      }
+
+      return errorsPlanUpgrade;
+    }
+
+    function isValidUpgradeablePlan(){
+      var haveErrors = (vm.errorsPlanUpgrade.isDKIMError || vm.errorsPlanUpgrade.isDeliveredError || vm.errorsPlanUpgrade.isDeliveredAndDKIMError)? true : false;
+      return !haveErrors;
+    }
+
+    $scope.disableNotUpgradable=function(event){
+      if(vm.isValidUpgradeablePlan){
+        return true;
+      }
+      else{
+        event.preventDefault();
```

Comment:
I think it's better to not allow the user to do an action before, rather stopping an action after doing it.

Reply:
if we want to use _DOM_, we could disable the button. If we want to use logic, we do not need to patch the event handling.

Reply:
@andresmoschini this was talked with @mvillaseco, currently the buttons are not really buttons, they are <a> tags being formatted to a button with css.

I apply this approach to prevent to change the html structure. To not used this I need to create new styles to prevent add bugs in styles to other elements that could be using the same approach using <a> elements.

Mauro recommend me to use pointer-events: none to avoid the js to prevent the action. But this could have problems with cross-platform compatibility following this table:
![git](https://user-images.githubusercontent.com/7245667/37907586-f289b014-30db-11e8-8494-ae4de7056c99.png)

Let me know what do you think about mantain this approach or change it for another one.

Thanks!

Reply:
UPDATE: I will implement current logic to disabled buttons instead to add a new inline validation.

Reply:
Thanks for your research @cristiangiagante, at the end we have talked and decided to try to reproduce the old approach, I mean, the same that we are doing to disable current plan to not allow to select it again.

![image](https://user-images.githubusercontent.com/1157864/37909380-189197d6-30e1-11e8-8105-b72d85b0c576.png)